src/pages/workspaces/workspace/analysis/modals/AnalysisDuplicator.test.ts

```diff
@@ -0,0 +1,351 @@
+import '@testing-library/jest-dom'
+
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { act } from 'react-dom/test-utils'
+import { h } from 'react-hyperscript-helpers'
+import { AzureStorage, AzureStorageContract } from 'src/libs/ajax/AzureStorage'
+import { GoogleStorage, GoogleStorageContract } from 'src/libs/ajax/GoogleStorage'
+import { defaultAzureWorkspace, defaultGoogleWorkspace } from 'src/pages/workspaces/workspace/analysis/_testData/testData'
+import {
+  AbsolutePath, AnalysisFile, getDisplayName, getExtension, getFileName
+} from 'src/pages/workspaces/workspace/analysis/file-utils'
+import {
+  AnalysisDuplicator,
+  AnalysisDuplicatorProps
+} from 'src/pages/workspaces/workspace/analysis/modals/AnalysisDuplicator'
+import { cloudProviders } from 'src/pages/workspaces/workspace/analysis/runtime-utils'
+import {
+  getToolFromFileExtension,
+  ToolLabel,
+  toolLabelTypes,
+  tools
+} from 'src/pages/workspaces/workspace/analysis/tool-utils'
+import { cloudProviderTypes } from 'src/pages/workspaces/workspace/workspace-utils'
+
+
+jest.mock('src/components/Modal', () => {
+  const { div, h } = jest.requireActual('react-hyperscript-helpers')
+  const originalModule = jest.requireActual('src/components/Modal')
+  return {
+    ...originalModule,
+    __esModule: true,
```

Comment:
What does the line above do?

Reply:
not 100% sure this is some common mocked code... might be time to start thinking about a `mockedCommons`, as this snippet is in 3 files now. Thoughts? cc @msilva-broad

Reply:
yes, it's on my list to spend some time on common mocks pattern that has good type safety and discoverability (jest's default implementation falls short on both counts last I checked).

as to __esModule: true, that is required for modules that export a default export

Reply:
Alternatively, adding a div with ID 'modal-root' removes the need to mock Modal. This could go in shared test setup.

```js
beforeAll(() => {
  const modalRoot = document.createElement('div')
  modalRoot.id = 'modal-root'
  document.body.append(modalRoot)
})

afterAll(() => {
  document.getElementById('modal-root')!.remove()
})
```